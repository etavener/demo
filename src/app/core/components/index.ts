export * from './card-form/card-form.component';
export * from './completed-message/completed-message.component';
export * from './payment-info/payment-info.component';
export * from './progress-display/progress-display.component';
export * from './input-expiry-date/input-expiry-date.component';
