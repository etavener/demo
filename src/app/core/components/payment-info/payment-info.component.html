<div class="payment-info">
  <div class="payment-info__value">
    <!-- todo : make a pipe to render value as Â£**.** using amount and currency -->
    <span>{{ details.value.amount }}</span>
    <span>&nbsp;</span>
    <span>{{ details.value.currency }}</span>
  </div>
  <div class="payment-info__description">
    <span>{{ details.description }}</span>
  </div>
</div>
