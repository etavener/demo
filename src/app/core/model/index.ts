export * from './payment';
export * from './worldpayResponse';
