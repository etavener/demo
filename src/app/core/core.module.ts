import { NgModule } from '@angular/core';
import {
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent,
} from './components';

import {
  DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE,
  MatButtonModule,
  MatDatepickerModule,
  MatDividerModule,
  MatFormFieldModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatSelectModule,
  MatStepperModule
} from '@angular/material';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule} from '@angular/router';
import { STEPPER_GLOBAL_OPTIONS } from '@angular/cdk/stepper';
import { HttpClientModule } from '@angular/common/http';
import { MomentDateAdapter } from '@angular/material-moment-adapter';
import {WorldpayService} from './services/worldpay.service';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';

export const MY_FORMATS = {
  parse: {
    dateInput: 'MM/YYYY',
  },
  display: {
    dateInput: 'MM/YYYY',
    monthYearLabel: 'MMM YYYY',
    dateA11yLabel: 'LL',
    monthYearA11yLabel: 'MMMM YYYY',
  },
};

const DECLARATIONS = [
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent
];

@NgModule({
  declarations: [
    ...DECLARATIONS
  ],
  exports: [
    ...DECLARATIONS
  ],
  imports: [
    FormsModule,
    ReactiveFormsModule,
    RouterModule,
    MatStepperModule,
    MatIconModule,
    MatDatepickerModule,
    MatInputModule,
    MatFormFieldModule,
    MatProgressSpinnerModule,
    MatSelectModule,
    MatButtonModule,
    MatDividerModule,
    HttpClientModule,
    BrowserAnimationsModule
  ],
  providers: [
    WorldpayService,
    {
      provide: STEPPER_GLOBAL_OPTIONS,
      useValue: { displayDefaultIndicatorType: false }
    },
    {
      provide: DateAdapter,
      useClass: MomentDateAdapter,
      deps: [MAT_DATE_LOCALE]},
    {
      provide: MAT_DATE_FORMATS,
      useValue: MY_FORMATS
    },
  ]
})
export class CoreModule { }
