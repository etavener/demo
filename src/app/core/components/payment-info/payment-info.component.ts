import { Component, Input } from '@angular/core';
import { PaymentDetails } from '../../model';

@Component({
  selector: 'payment-info',
  templateUrl: './payment-info.component.html',
  styleUrls: ['./payment-info.component.scss']
})
export class PaymentInfoComponent {

  @Input() details: PaymentDetails;

  constructor() { }

}
