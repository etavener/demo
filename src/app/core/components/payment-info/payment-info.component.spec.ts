import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { PaymentInfoComponent } from './payment-info.component';
import {Component} from '@angular/core';

describe('Payment Info Component', () => {
  let fixture: ComponentFixture<TestPaymentInfoComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        PaymentInfoComponent,
        TestPaymentInfoComponent
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestPaymentInfoComponent);
    fixture.detectChanges();
  });

  it('should render the content', () => {
    expect(fixture).toMatchSnapshot();
  });

  @Component({
    selector: 'test-payment-info',
    template: `<payment-info [details]="details"></payment-info>`
  })
  class TestPaymentInfoComponent {
    details = {
      description: 'description',
      value: {
        currency: 'GBP',
        amount: 100
      }
    };
  }

});
