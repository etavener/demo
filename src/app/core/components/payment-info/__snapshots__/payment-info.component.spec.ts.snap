// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Info Component should render the content 1`] = `
<test-payment-info
  details={[Function Object]}
>
  <payment-info
    ng-reflect-details="[object Object]"
  >
    <div
      class="payment-info"
    >
      <div
        class="payment-info__value"
      >
        <span>
          100
        </span>
        <span>
          Â 
        </span>
        <span>
          GBP
        </span>
      </div>
      <div
        class="payment-info__description"
      >
        <span>
          description
        </span>
      </div>
    </div>
  </payment-info>
</test-payment-info>
`;
